server/rest/src/main/java/org/infinispan/rest/resources/ProtobufResource.java

```diff
@@ -0,0 +1,178 @@
+package org.infinispan.rest.resources;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.infinispan.AdvancedCache;
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.infinispan.query.remote.ProtobufMetadataManager;
+import org.infinispan.query.remote.client.ProtobufMetadataManagerConstants;
+import org.infinispan.rest.InvocationHelper;
+import org.infinispan.rest.NettyRestResponse;
+import org.infinispan.rest.cachemanager.RestCacheManager;
+import org.infinispan.rest.framework.ContentSource;
+import org.infinispan.rest.framework.ResourceHandler;
+import org.infinispan.rest.framework.RestRequest;
+import org.infinispan.rest.framework.RestResponse;
+import org.infinispan.rest.framework.impl.Invocations;
+import org.infinispan.rest.operations.exceptions.NoDataFoundException;
+import org.infinispan.rest.operations.exceptions.NoKeyException;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_JSON;
+import static org.infinispan.rest.framework.Method.DELETE;
+import static org.infinispan.rest.framework.Method.GET;
+import static org.infinispan.rest.framework.Method.POST;
+import static org.infinispan.rest.framework.Method.PUT;
+
+/**
+ * Protobuf schema manipulation Resource
+ *
+ * @author Katia Aresti
+ * @since 11
+ */
+public class ProtobufResource extends BaseCacheResource implements ResourceHandler {
+
+   public ProtobufResource(InvocationHelper invocationHelper) {
+      super(invocationHelper);
+   }
+
+   @Override
+   public Invocations getInvocations() {
+      return new Invocations.Builder()
+            // Key related operations
+            .invocation().methods(GET).path("/v2/protobuf").handleWith(this::getSchemasNames)
```

Comment:
I am conflicted between ```/v2/protobuf``` or ```/v2/schema``` for the REST endpoint path. WDYT @anistor?

Reply:
@gustavonalle I know the question isn't for me but I'd be in favour of `/v2/schema` for consistency with the CLI as that is the command you use to upload .proto files.

Reply:
Allright!!!